client/packages/plugins/README.md

```diff
@@ -28,233 +28,123 @@ When running in development mode, the required plugin files are loaded directly
 
-```typescript
-  const columnDefinitions: ColumnDescription<StockLineRowFragment>[] = [
-    [standard column definitions go here],
-    ...pluginColumns,
-  ];
-
-...
-
-  const columns = useColumns<StockLineRowFragment>(
-    columnDefinitions,
-    {
-      sortBy,
-      onChangeSortBy: updateSortQuery,
-    },
-    [sortBy, pluginColumns]
-  );
-```
+### Plugin data
 
+TODO update this
 
-### Events
+Plugins can store data in the `plugin_data` table. The following methods are available in the graphQL API for interacting with plugin data:
 
-Events can be used to interact between the app and a plugin.
-To register for an event use the `usePluginEvents` hook.
+- `pluginData`
+- `insertPluginData`
+- `updatePluginData`
 
-For example, here is how you can call the `onSave` method within a plugin, when the `onSaveStockEditForm` event is dispatched:
+The querying and mutating of data follows the standard pattern used throughout open mSupply:
 
 ```typescript
-  const { addEventListener, removeEventListener, dispatchEvent } = usePluginEvents();
-
-...
-
-   useEffect(() => {
-    const listener: PluginEventListener = {
-      eventType: 'onSaveStockEditForm', // see the event types for available options
-      listener: onSave, // provide a method here with no parameters. This method will be called when plugins dispatch onSaveStockEditFormEvent, see example below
-    };
-    addEventListener(listener);
-
-    return () => removeEventListener(listener); // remove the event listener when the component is unmounted
-  }, [onSave, addEventListener, removeEventListener]);
+  const { data } = usePluginData.data(stockLine?.id ?? '');
+  const { mutate } = data?.id ? usePluginData.update() : usePluginData.insert();
 ```
 
-In the above code, the plugin is listening for an `onSave` event within the `StockEditForm` (which has a specific event type of `onSaveStockEditForm`).
-When that event is dispatched, the plugin responds by calling its `onSave` method.
-
-The corresponding code in the `StockLineEditModal` looks like this:
-
-```typescript
-const { dispatchEvent, addEventListener, removeEventListener } = usePluginEvents();
-
-// and then, the `dispatchEvent` call is added to the save button's onClick method:
-onClick={() =>
-    getConfirmation({
-      onConfirm: async () => {
-        await onSave();
-        dispatchEvent('onSaveStockEditForm', new Event(draft.id));
-        onClose();
-      },
-    })
-  }
-```
+These functions can be implemented within your plugin and used to fetch and update data.
 
-Events can be dispatched and handled in the other direction too - raised by a plugin, and responded to within the main application.
-For example, here is how an edit form can trigger the validation method when the plugin data is changed:
 
-```typescript
-  const { dispatchEvent, addEventListener, removeEventListener } = usePluginEvents();
+## Creating a plugin
```

Comment:
Would be good to mention in this section that hot module reloading is available for local dev, but frontend will need to be restarted once on initial addition of the plugin bundle submodule/folder

Reply:
a2aee2c0fbe71a96883c9aa465589e657c0eef65